documentation/getting-started.adoc

```diff
@@ -0,0 +1,245 @@
+== Getting started
+
+=== Prerequisites
+
+Kubernetes or OpenShift cluster are required to deploy Strimzi. Strimzi supports all kind of clusters - from public and
+private clouds up to local deployments intended for development purposes. This guide expects that a Kubernetes or
+OpenShift cluster is available and the `kubectl` or `oc` command line tools are installed and configured to connect
+to the running cluster.
+
+When no Kubernetes or OpenShift cluster is available, `Minikube` or `Minishift` can be used. More details can be found
+in <<_installing_kubernetes_and_openshift_cluster>>
+
+
+=== Cluster Controller
+
+Strimzi uses a component called the Cluster Controller to deploy and manage Kafka (including Zookeeper) and Kafka Connect
+clusters. The Cluster Controller is deployed as a process running inside your Kubernetes or OpenShift cluster. To deploy a
+Kafka cluster, a ConfigMap with the cluster configuration has to be created. Based on the information in that ConfigMap,
+the Cluster Controller will deploy a corresponding Kafka cluster. By default, the ConfigMap needs to be labeled with
+following labels:
+
+[source,yaml]
+strimzi.io/type: kafka
+strimzi.io/kind: cluster
+
+and contain the cluster configuration in a specific format.
+
+The Strimzi project contains example YAML files which make deploying of Cluster Controller easier.
+
+==== Deploying to Kubernetes
+
+To deploy the Cluster Controller on Kubernetes, following command should be executed:
+
+[source]
+kubectl create -f resources/kubernetes/cluster-controller.yaml
+
+To verify whether the Cluster Controller has been deployed successfully, the Kubernetes Dashboard or the following
+command can be used:
+
+[soruce]
+kubectl describe all
+
+==== Deploying to OpenShift
+
+To deploy the Cluster Controller on OpenShift, following command should be executed:
+
+[source]
+oc create -f resources/openshift/cluster-controller-with-template.yaml
+
+To verify whether the Cluster Controller has been deployed successfully, the OpenShift console or the following command
+can be used:
+
+[soruce]
+oc describe all
+
+=== Kafka broker
+
+Strimzi uses StatefulSets feature to deploy Kafka brokers on Kubernetes/OpenShift.
+With StatefulSets, the pods receive a unique name and network identity and that makes it easier to identify the
+individual Kafka broker pods and set their identity (broker ID). The deployment uses **regular** and **headless**
+services:
+
+- regular services can be used as bootstrap servers for Kafka clients
+- headless services are needed to have DNS resolve the pods IP addresses directly
+
+Together with Kafka, Strimzi also installs a Zookeeper cluster and configures the Kafka brokers to connect to it. The
```

Comment:
"As well as Kafka, ..." would be better than "Together with..." because could be interpreted to mean we somehow deploy Zookeeper on the same nodes as the Kafka brokers.